helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -706,54 +732,77 @@ public Builder heliosClient(final HeliosClient heliosClient) {
-     * {@link HeliosSoloDeployment} created with this Builder. If unset a random string will be
-     * used.
-     *
-     * @param username The Helios user to identify as.
-     * @return This Builder, with its Helios username configured.
+     * This method exists only for testing. We need to mock the logic that probes the master port.
+     * @param soloMasterProber A {@link SoloMasterProber}
+     * @return This Builder, with its {@link SoloMasterProber} configured.
      */
-    public Builder heliosUsername(final String username) {
-      this.heliosUsername = username;
+    @VisibleForTesting
+    Builder soloMasterProber(final SoloMasterProber soloMasterProber) {
+      this.soloMasterProber = soloMasterProber;
       return this;
     }
 
     /**
-     * Optionally specify an environment variable to be set inside the Helios solo container.
-     * @param key Environment variable to set.
-     * @param value Environment variable value.
-     * @return This Builder, with the environment variable configured.
+     * This method exists only for testing. We need to mock the logic checking the solo agent is up.
+     * @param soloAgentProber A {@link SoloAgentProber}
+     * @return This Builder, with its {@link SoloAgentProber} configured.
      */
-    public Builder env(final String key, final Object value) {
-      this.env.add(key + "=" + value.toString());
+    @VisibleForTesting
+    Builder soloAgentProber(final SoloAgentProber soloAgentProber) {
+      this.soloAgentProber = soloAgentProber;
+      return this;
+    }
+
+
+    /**
+     * This method exists only for testing.
+     * We need to mock the logic to connect to the solo watchdog.
+     * @param soloWatchdogConnector A {@link SoloWatchdogConnector}
+     * @return This Builder, with its {@link SoloWatchdogConnector} configured.
+     */
+    @VisibleForTesting
+    Builder soloWatchdogConnector(final SoloWatchdogConnector soloWatchdogConnector) {
+      this.soloWatchdogConnector = soloWatchdogConnector;
+      return this;
+    }
+
+    /**
+    * This method exists only for testing.
+    * For unit tests that check we forcibly kill helios-solo if the timeout period for it to exit
+    * by itself has elapsed, we want to set a shorter timeout.
+    * @param exitTimeoutSeconds Number of seconds to wait until we call docker kill on helios-solo.
+    * @return This Builder, with its exit timeout configured.
+    */
+    @VisibleForTesting
+    Builder soloExitTimeoutSeconds(final long exitTimeoutSeconds) {
+      this.exitTimeoutSeconds = exitTimeoutSeconds;
       return this;
     }
 
     /**
      * Configures, deploys, and returns a {@link HeliosSoloDeployment} using the as specified by
      * this Builder.
      *
-     * @return A Helios Solo deployment configured by this Builder.
+     * @return A HeliosSoloDeployment configured by this Builder.
      */
-    public HeliosSoloDeployment build() {
+    public HeliosDeployment build() {
```

Comment:
declaring that the `build()` method returns a superclass isn't really useful at all since the class is `HeliosSoloDeployment.Builder`, and it's not like the `Builder` itself has any parent classes where someone might refer to it as `HeliosDeployment.Builder` etc. 

Worst case this type of change forces code using this to have to cast the `build()` result to `(HeliosSoloDeployment)` which is just unnecessary. 

(also I am a little sour since this undoes something I just did in https://github.com/spotify/helios/commit/144600b120bd095dd2882f0dd63a5a065949056e )